slatedb/src/db_iter.rs

```diff
@@ -123,39 +125,23 @@ impl<'a> KeyValueIterator for GetIterator<'a> {
         Ok(())
     }
 
-    async fn next(&mut self) -> Result<Option<KeyValue>, SlateDBError> {
```

Comment:
I moved this into `next_entry` -- no reason not to have it there in first place 🤔 and it was causing some bugs because `next` is no longer directly called on this iterator (its called on the `MergeRequiredIterator` or the `MergeOperatorIterator`)